### PR TITLE
Add capture support for VK_ANDROID_frame_boundary

### PR DESCRIPTION
**Pull request by mikes-lunarg**
Capture the vkFrameBoundaryANDROID function and treat it as a frame delimiter.

It is very simple extension. The most complicated parts are:
- There was an existing bug in the capture layer where it wouldn't always return an pointer for layer-implemented functions. This is handled in the first commit.
- VK_ANDROID_frame_boundary is not in the main spec so this change also adds a way to patch the vk.xml tree so the code generators can know about the extension.

Questions:
- **Is this useful?** I thinks so, HWUI is already using it. Ideally this private vendor extension would be replaced with a EXT or KHR extension and hosted in the main spec.
- **Should there be an option to enable it?** Probably yes. It makes sense to make undocumented/experimental functionality opt-in.
- **Should it be exposed on Android only?** I'm not sure. Exposing it on other platforms doesn't hurt, but also isn't very useful.
- **Should it be exposed in the layer manifest file?** Probably no. This is a private extension so I don't see much value in telling the desktop loader we support it.

Future work:
- It would be nice to hook this up to the virtual swapchain (and screenshot!) on replay. Replaying a blank screen isn't very exciting...

### CODE REVIEW

File: framework/format/VK_ANDROID_frame_boundary.h [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+** FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+** DEALINGS IN THE SOFTWARE.
+*/
+
+#ifndef GFXRECON_VK_ANDROID_FRAME_BOUNDARY
+#define GFXRECON_VK_ANDROID_FRAME_BOUNDARY
+
+#include "vulkan/vulkan.h"
+
+typedef void(VKAPI_PTR* PFN_vkFrameBoundaryANDROID)(VkDevice device, VkSemaphore semaphore, VkImage image);
```
Comment (bradgrantham-lunarg) [inline framework/format/VK_ANDROID_frame_boundary.h:28]: Is there a reason this couldn't live in framework/format/platform_types.h with the other Vulkan functions?

Comment (bradgrantham-lunarg) [reply framework/format/VK_ANDROID_frame_boundary.h:28]: Oh, I see you included this from there.  :)  Interesting.  I'll let that soak for a day or two.

**Review by bradgrantham-lunarg — APPROVED** [review]
~You may have a conflict with api IDs added by #980~
Yeah, I think that's what the conflicts listed below are.

**Comment by MarkY-LunarG** [comment]
Closing and re-opening as PR #1288

**Pull request closed without merging**